templates/vpn.yaml.tmpl

```diff
@@ -0,0 +1,140 @@
+        [Install]
+        WantedBy=multi-user.target
+    - name: template-ipsec-config.service
+      enabled: true
+      contents: |
+        [Unit]
+        Description=oneshot command for filling ip addreses into ipsec configs
+        Wants=network-online.target
+        After=network-online.target
+        Before=strongswan.service
+
+        [Service]
+        Type=oneshot
+        ExecStart=/opt/bin/template-ipsec-config
+
+        [Install]
+        WantedBy=multi-user.target
+    - name: setup-iptables.service
+      enabled: true
+      contents: |
+        [Unit]
+        Description=oneshot command for setup iptables rules for ipsec
+        Wants=network-online.target template-ipsec-config.service
+        After=network-online.target template-ipsec-config.service
+
+        [Service]
+        Type=oneshot
+        ExecStart=/opt/bin/setup-iptables
+
+        [Install]
+        WantedBy=multi-user.target
+    - name: strongswan.service
+      enabled: true
+      contents: |
+        [Unit]
+        Description=strongswan ipsec service running in docker
+        Requires=docker.service template-ipsec-config.service
+        After=docker.service template-ipsec-config.service
+
+        [Service]
+        Environment="NAME=%p.service"
+        Environment="IMAGE={{.DockerRegistry}}/giantswarm/docker-strongswan:d432793577e07bfe6856d3739e6bca3d91db71e4"
+        Restart=always
+        RestartSec=10
+        TimeoutStopSec=10
+        ExecStartPre=/usr/bin/docker pull $IMAGE
+        ExecStartPre=/bin/bash -c "touch /var/log/charon.log"
+        ExecStart=/usr/bin/docker run --rm --net=host --privileged \
+                  -v /etc/strongswan/vpn-config.conf:/etc/strongswan/ipsec.d/connections/vpn-config.conf \
+                  -v /etc/strongswan/ipsec.secrets:/etc/strongswan/ipsec.secrets \
+                  -v /var/log/charon.log:/var/log/charon.log \
+                  -v /usr/sbin/modprobe:/usr/sbin/modprobe \
+                  --name $NAME $IMAGE 
+        ExecStop=-/usr/bin/docker stop -t 10 $NAME
+
+        [Install]
+        WantedBy=multi-user.target
+storage:
+  files:
+    - path: /opt/bin/template-ipsec-config
+      filesystem: root
+      mode: 0700
+      contents:
+        inline: |
+          #!/bin/bash 
+          while ! curl --connect-timeout 5 http://169.254.169.254 >/dev/null; do sleep 1s; echo "waiting for vpc endpoint"; done;
+
+          PRIVATE_IP=$(curl http://169.254.169.254/latest/meta-data/local-ipv4) 
+          PUBLIC_IP=$(curl http://169.254.169.254/latest/meta-data/public-ipv4)
+
+          sed -i "s/__PRIVATE_IP__/${PRIVATE_IP}/g" /etc/strongswan/vpn-config.conf /etc/strongswan/ipsec.secrets /opt/bin/setup-iptables
+          sed -i "s/__PUBLIC_IP__/${PUBLIC_IP}/g" /etc/strongswan/vpn-config.conf /etc/strongswan/ipsec.secrets
+
+          echo "Files /etc/strongswan/vpn-config.conf /etc/strongswan/ipsec.secrets succesfully templated."
```

Comment:
templating ipsec config and iptable rules, as this configs require the private and public ip info we don't have at time when terraform ignition is rendered